modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2250,9 +2256,12 @@ private CompletableFuture<ReplicaResult> processMultiEntryAction(ReadWriteMultiR
 
                         boolean insert = rowId == null;
 
-                        RowId rowId0 = insert ? new RowId(partId(), UUID.randomUUID()) : rowId;
+                        RowId rowId0 = insert
+                                ? newKeyMap.computeIfAbsent(pk.byteBuffer(), rowIdGen)
+                                : rowId;
 
                         if (lastCommitTime != null) {
+                            assert rowId != null : "lastCommitTime is not null, but rowId is null";
```

Comment:
This assert looks unrelated to bug. This is executed only for existing rows.

Reply:
It's just to fix the IDE warning. Replaced it with `noinspection`.